src/mono/mono/eventpipe/ep-rt-mono.c

```diff
@@ -2042,10 +2708,36 @@ EventPipeEtwCallbackDotNETRuntime (
 			mono_profiler_set_jit_failed_callback (_ep_rt_mono_profiler, profiler_jit_failed);
 			mono_profiler_set_jit_done_callback (_ep_rt_mono_profiler, profiler_jit_done);
 			mono_profiler_set_image_loaded_callback (_ep_rt_mono_profiler, profiler_image_loaded);
+			mono_profiler_set_image_unloaded_callback (_ep_rt_mono_profiler, profiler_image_unloaded);
 			mono_profiler_set_assembly_loaded_callback (_ep_rt_mono_profiler, profiler_assembly_loaded);
+			mono_profiler_set_assembly_unloaded_callback (_ep_rt_mono_profiler, profiler_assembly_unloaded);
+			mono_profiler_set_thread_started_callback (_ep_rt_mono_profiler, profiler_thread_started);
+			mono_profiler_set_thread_stopped_callback (_ep_rt_mono_profiler, profiler_thread_stopped);
+			mono_profiler_set_class_loading_callback (_ep_rt_mono_profiler, profiler_class_loading);
+			mono_profiler_set_class_failed_callback (_ep_rt_mono_profiler, profiler_class_failed);
+			mono_profiler_set_class_loaded_callback (_ep_rt_mono_profiler, profiler_class_loaded);
+			mono_profiler_set_exception_throw_callback (_ep_rt_mono_profiler, profiler_exception_throw);
+			mono_profiler_set_exception_clause_callback (_ep_rt_mono_profiler, profiler_exception_clause);
+			mono_profiler_set_monitor_contention_callback (_ep_rt_mono_profiler, profiler_monitor_contention);
```

Comment:
Can we avoid setting a profiler callback for some of these events unless we know the event is enabled?  Or is it too late by then?

Reply:
Its too late, we would then need to move the check to the places where mono profiler fires its events. The callback is only set/unset if we have any code interested in the native runtime provider, if not no callbacks will be set.

Reply:
ah ok, I misunderstood what the "enabled" was meant - It's the native runtime provider, not EP/DS in general.  That's alright, then.